src/jit/simdcodegenxarch.cpp

```diff
@@ -840,13 +1004,548 @@ void CodeGen::genSIMDIntrinsicUnOp(GenTreeSIMD* simdNode)
+        getEmitter()->emitIns_R_I(INS_psrlq, emitActualTypeSize(simdType), tmpReg2, 32);
+        getEmitter()->emitIns_R_I(INS_mov, EA_8BYTE, tmpIntReg, 0X4530000000000000);
+        inst_RV_RV(INS_mov_i2xmm, tmpReg, tmpIntReg, TYP_ULONG);
+        if (iset == InstructionSet_AVX)
+        {
+            inst_RV_RV(INS_vpbroadcastq, tmpReg, tmpReg, simdType, emitActualTypeSize(simdType));
+        }
+        else
+        {
+            inst_RV_RV(INS_movlhps, tmpReg, tmpReg, simdType, emitActualTypeSize(simdType));
+        }
+        inst_RV_RV(INS_orpd, targetReg, tmpReg, simdType, emitActualTypeSize(simdType));
+        inst_RV_RV(INS_subpd, targetReg, tmpReg, simdType, emitActualTypeSize(simdType));
+        getEmitter()->emitIns_R_I(INS_mov, EA_8BYTE, tmpIntReg, 0X4330000000000000);
+        inst_RV_RV(INS_mov_i2xmm, tmpReg, tmpIntReg, TYP_ULONG);
+        if (iset == InstructionSet_AVX)
+        {
+            inst_RV_RV(INS_vpbroadcastq, tmpReg, tmpReg, simdType, emitActualTypeSize(simdType));
+        }
+        else
+        {
+            inst_RV_RV(INS_movlhps, tmpReg, tmpReg, simdType, emitActualTypeSize(simdType));
+        }
+        inst_RV_RV(INS_orpd, tmpReg2, tmpReg, simdType, emitActualTypeSize(simdType));
+        inst_RV_RV(INS_subpd, tmpReg2, tmpReg, simdType, emitActualTypeSize(simdType));
+        inst_RV_RV(INS_addpd, targetReg, tmpReg2, simdType, emitActualTypeSize(simdType));
+    }
+    else if ((intrinsicID == SIMDIntrinsicConvertToDouble) && (baseType == TYP_LONG))
+    {
+        instruction rightShiftIns = getOpForSIMDIntrinsic(SIMDIntrinsicShiftRightInternal, TYP_SIMD16);
+        instruction leftShiftIns  = getOpForSIMDIntrinsic(SIMDIntrinsicShiftLeftInternal, TYP_SIMD16);
+
+        if (iset == InstructionSet_AVX)
+        {
+            // Extract the high 16-bits
+            getEmitter()->emitIns_R_R_I(INS_vextractf128, EA_32BYTE, tmpReg, op1Reg, 0x01);
+
+            // Put v[3] (the high-order element) in tmpReg2 and convert it.
+            // TODO-CQ: Note that if we supported the two-reg form of shift, we wouldn't need to copy it here.
+            inst_RV_RV(ins_Copy(simdType), tmpReg2, tmpReg, simdType, emitActualTypeSize(simdType));
+            getEmitter()->emitIns_R_I(rightShiftIns, emitActualTypeSize(simdType), tmpReg2, 8);
+            genSIMDLo64BitConvert(intrinsicID, simdType, baseType, tmpReg2, tmpIntReg, tmpReg2);
+
+            // Now, shift the resulting 64-bits left.
+            getEmitter()->emitIns_R_I(leftShiftIns, emitActualTypeSize(simdType), tmpReg2, 8);
+
+            // Convert v[2], in the lo bits of tmpReg.
+            // For the convert to double, the convert preserves the upper bits in tmpReg2.
+            // For the integer convert, we have to put it in tmpReg and or it in, since movd clears the upper bits.
+            genSIMDLo64BitConvert(intrinsicID, simdType, baseType, tmpReg, tmpIntReg, tmpReg2);
```

Comment:
>genSIMDLo64BitConvert(intrinsicID, simdType, baseType, tmpReg, tmpIntReg, tmpReg2); [](start = 12, length = 83)

This routine doesn't work for x86 RyuJIT since it assumes 64-bit integer registers.

Please note that we have enabled SIMD even on x86 RyuJIT.

It is fine to stage the work: enable these intrinsics only for 64-bit and file a tracking issue to enable them for x86 RyuJIT.  Subsequently we can do the incremental work required for x86.